recipy/log.py

```diff
@@ -169,7 +172,8 @@ def log_output(filename, source):
 
     db = open_or_create_db()
 
-    if option_set('data', 'file_diff_outputs') and os.path.isfile(filename):
+    if option_set('data', 'file_diff_outputs') and os.path.isfile(filename) \
+       and not is_binary(filename):
```

Comment:
`if all(option_set('data', 'file_diff_outputs'), os.path.isfile(filename), not is_binary(filename)):`

Just about fits into a single line, not sure which version of python all() was introduced it certainly is in 2.6